backend/sirius-components-diagrams-layout/src/test/java/org/eclipse/sirius/components/diagrams/layout/services/DefaultTestDiagramDescriptionProvider.java

```diff
@@ -180,7 +184,27 @@ public DefaultTestDiagramDescriptionProvider(IObjectService objectService) {
         this.objectService = Objects.requireNonNull(objectService);
     }
 
+    private void populateTargetObjectIdToNodeId(List<Node> nodes, Map<String, String> targetObjectIdToNodeId) {
+        Optional<Node> optionalNode = Optional.empty();
+        List<Node> deeperNode = new ArrayList<>();
+
+        Iterator<Node> nodeIt = nodes.iterator();
+        while (optionalNode.isEmpty() && nodeIt.hasNext()) {
+            Node node = nodeIt.next();
+            targetObjectIdToNodeId.put(node.getTargetObjectId(), node.getId());
+            deeperNode.addAll(node.getChildNodes());
+            deeperNode.addAll(node.getBorderNodes());
+        }
+
+        if (!deeperNode.isEmpty()) {
+            this.populateTargetObjectIdToNodeId(deeperNode, targetObjectIdToNodeId);
+        }
+    }
+
     private Function<VariableManager, List<org.eclipse.sirius.components.representations.Element>> getEdgeTargetProvider(Diagram diagram) {
+        Map<String, String> targetObjectIdToNodeId = new HashMap<>();
+        this.populateTargetObjectIdToNodeId(diagram.getNodes(), targetObjectIdToNodeId);
```

Comment:
The method could simply return the Map<String, String> and create it itself. It would probably simplify what you are trying to do.

Reply:
Done